src/main/sensors/battery.c

```diff
@@ -159,6 +159,11 @@ void batteryUpdatePresence(void)
     
     bool isVoltageStable = ABS(voltageMeter.filtered - previousVoltage) <= VBAT_STABLE_MAX_DELTA;
 
+#ifdef USE_ESC_SENSOR
+    // Delaying cell detection when voltage meter soource is ESC Sensor
+    isVoltageStable = (batteryConfig()->voltageMeterSource == VOLTAGE_METER_ESC && millis() > ESC_BOOTTIME);
```

Comment:
I would make this `isVoltageStable = isVoltageStable && ((batteryConfig()->voltageMeterSource == VOLTAGE_METER_ESC && millis() > ESC_BOOTTIME))`. No point *not* checking for the voltage being stable.